Application.cpp

```diff
@@ -282,20 +285,21 @@ bool Application::initWindowAndDevice() {
 	requiredLimits.limits.maxBindGroups = 2;
 	requiredLimits.limits.maxUniformBuffersPerShaderStage = 2;
 	requiredLimits.limits.maxUniformBufferBindingSize = 16 * 4 * sizeof(float);
-	// Allow textures up to 2K
-	requiredLimits.limits.maxTextureDimension1D = 2048;
-	requiredLimits.limits.maxTextureDimension2D = 2048;
 	requiredLimits.limits.maxTextureArrayLayers = 1;
 	requiredLimits.limits.maxSampledTexturesPerShaderStage = 1;
 	requiredLimits.limits.maxSamplersPerShaderStage = 1;
 
+	// Always use adapter's texture resulution limits.
+	requiredLimits.limits.maxTextureDimension1D = supportedLimits.limits.maxTextureDimension1D;
+	requiredLimits.limits.maxTextureDimension2D = supportedLimits.limits.maxTextureDimension2D;
+	requiredLimits.limits.maxTextureDimension3D = supportedLimits.limits.maxTextureDimension3D;
```

Comment:
Not sure about this one, it is handy but the general policy of the guide is "request everything you need" because a typical advanced scenario would be there's a tier1/tier2/tier3 set of limits that determines the "quality" of the device wrt the end application. That being said, texture size may be treated a little different idk. Let's leave it aside for now.

Reply:
In order to support creating textures the size of the surface's resolution (4k), we need to use adapter's limits. It's needed atleast for wgpu.

Reply:
True, I'm guilty of not having any screen bigger than HD in my daily life ^^ I'll think about it, though another option would be to just bump it to 4096. If I don't set it to a fixed value I'll have to explain why in the guide.